### PR TITLE
Move create temp login logic to it's own function

### LINKED ISSUE #1482: Setting password via /setpassword signs in user. Does not generate re-authentication credentials.
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

view the job on Upwork [here](https://www.upwork.com/jobs/~01ccfce466ee66ebaf).

**Platform - version:**  
Current

# Problem
It doesn't seem like we are generating re-authentication credentials when a user hits `https://expensify.cash/setpassword/[validateCode]`.

# Why is this important?
I think anyone setting a password will be logged out when their authToken expires. And ideally they won't be logged out at all once a password is set.

# Solution
Generalize the logic [here](https://github.com/Expensify/ReactNativeChat/blob/5739dc888092440ebfc1463fe47db6e7159f90b8/src/libs/actions/Session.js#L127-L165) into a private action that can be used in the `signIn()` and `setPassword()` flows.

### PR DESCRIPTION
**Pull request by anthony-hull**
Call createTemporaryLogin in the set password function to allow token refresh after exp
Change redirect check to look for newly named generated password key in ONYX

### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1482

### Tests

1. Trigger password reset on expensify.com for your account.
2. Make sure you are logged out.
3. Navigate to: http://expensify.cash/#/setpassword/[code from magic link in email here]
4. Make sure you have credentials.login key set to your email address in local storage.
5. Set a new password.
6. Edit the session.authtoken in local storage and deface it somehow.
7. if you have any photos in the chat log, they will disappear.
8. Do any action that initiates an API call.
9. Check and see that the auth token has been replaced and the app works normally.
10. Navigate to http://expensify.cash/#/setpassword/[any junk here]
11. See how you're sent to the root of the app.

This time follow steps 6-11 but after logging in normally to ensure previous functionality is still working.

### Tested On

- [X] Web
- [X] Mobile Web
- [ ] iOS
- [ ] Desktop
- [ ] Android

### Screenshots
I didn't touch any presentational logic or stylings, there isn't anything to screenshot.

### CODE REVIEW

**Comment by thienlnam** [comment]
Code looks fine but not familiar with logic stuff so ccing in @marcaaron

File: src/libs/actions/Session.js [hunk 1]
```diff
@@ -124,46 +177,7 @@ function signIn(password, twoFactorAuthCode) {
...
-
-                    Onyx.merge(ONYXKEYS.CREDENTIALS, {autoGeneratedLogin, autoGeneratedPassword});
-                })
-                .catch((error) => {
-                    Onyx.merge(ONYXKEYS.ACCOUNT, {error: error.message});
-                })
-                .finally(() => {
-                    Onyx.merge(ONYXKEYS.ACCOUNT, {loading: false});
-                });
+            createTemporaryLogin(authenticateResponse);
```
Comment (marcaaron) [inline src/libs/actions/Session.js:180]: NAB, if we only need the `authToken` and `email` maybe we can pass these are separate parameters to `createTemporaryLogin()`.

**Review by marcaaron — COMMENTED** [review]
LGTM, @thienlnam did you test the flows as well?

**Comment by marcaaron** [comment]
@anthony-hull I'm noticing that the resend validation link flow is not working correctly at the moment (which is probably why you are testing this on web). Would you mind adding the command `ResendValidateCode` to this list [here](https://github.com/Expensify/Expensify.cash/blob/ee94bc33980bb7b2cfcb0d2932c16514e8c080e1/src/libs/API.js#L30-L36)? This should fix the bug where tons of errors appear in the console.

File: src/libs/actions/Session.js (hunk 1 above)
Comment (anthony-hull) [reply src/libs/actions/Session.js:180]: NAB? 
I'll fix that :)

Comment (marcaaron) [reply src/libs/actions/Session.js:180]: Ah sorry this is shorthand for "not a blocker" but will make someone happy if you decide to fix

**Comment by marcaaron** [comment]
Tested these changes and they are working well.

**Review by marcaaron — DISMISSED** [review]

**Comment by anthony-hull** [comment]
I'll do those two changes now

File: src/libs/actions/Session.js [hunk 2]
```diff
@@ -103,6 +103,59 @@ function fetchAccountDetails(login) {
         });
 }
 
+/**
+ *
+ * Will create a temporary login for the user in the passed authenticate response which is used when
+ * re-authenticating after an authToken expires.
+ *
+ * @param {string} authToken
+ * @param {string} email
```
Comment (marcaaron) [inline src/libs/actions/Session.js:112]: Can we please capitalize these params?

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Review by marcaaron — APPROVED** [review]

**Comment by botify** [comment]
@marcaaron looks like this was merged without passing tests. Please add a note explaining why this was done or remove the `Emergency` label if this is not an emergency.

**Comment by marcaaron** [comment]
Not too sure why the `Emergency` label kicked on for this. Maybe because the CLA agreement was skipped? But Anthony has worked with us before and signed in so did not thing this was an issue.

**Comment by anthony-hull** [comment]
I found it. There is a commit that isn't signed

**Comment by anthony-hull** [comment]
At least I think that is why. I'm assuming the bot checks that I have signed everything? they are all verified. But one is missing the green tick.

I know why that happened as well. It was me who made the commit. I will fix that in my setup. 
Do you want me to fix the signed code issue?

**Comment by marcaaron** [comment]
Ah thanks, but I think there's nothing for you to do here. Looks like all commits were signed the green check is added when all the actions successfully run.

Screen Shot 2021-02-22 at 11 25 30 AM

**Pull request merged**